crates/block-producer/src/mempool/mod.rs

```diff
@@ -485,4 +493,22 @@ impl Mempool {
     ) -> Result<mpsc::Receiver<MempoolEvent>, BlockNumber> {
         self.subscription.subscribe(chain_tip)
     }
+
+    /// Adds mempool stats to the current tracing span.
+    ///
+    /// Note that these are only visible in the OpenTelemetry context, as conventional tracing
+    /// does not track fields added dynamically.
+    fn inject_telemetry(&self) {
+        let span = tracing::Span::current();
+
+        span.set_attribute("mempool.transactions.total", self.transactions.len());
+        span.set_attribute("mempool.transactions.roots", self.transactions.num_roots());
+        span.set_attribute("mempool.accounts", self.state.num_accounts());
+        span.set_attribute("mempool.nullifiers", self.state.num_nullifiers());
+        span.set_attribute("mempool.output_notes", self.state.num_notes_created());
+        span.set_attribute("mempool.batches.pending", self.batches.num_pending());
+        span.set_attribute("mempool.batches.proven", self.batches.num_proven());
+        span.set_attribute("mempool.batches.total", self.batches.len());
+        span.set_attribute("mempool.batches.roots", self.batches.num_roots());
```

Comment:
nit (feel free to ignore): it is not immediately obvious what `mempool.transactions.roots` and `mempool.batches.roots` means. Maybe there is a better name for them?